### DISCUSSION TITLE
Kafkauser

### DISCUSSION THREAD

**Discussion by sreejesh123** [strimzi/strimzi-kafka-operator#6314] [Q&A]
Sorry may be repeated q. But wanted to clarify before I go back to management with strimzi as our Kafka alternative

Our setup 
- self signed cluster ca  to be used which comes out of box. Not changing to use company issued cert due to not having root ca/key 

- client ca and   Kafka listener certificates  uses  Certmanager created Google cas signed certification .   

The clarification needed - when Kafkauser manifest is run with tls as option,  which certificate is used to Sign the user certificate ?  Is it client ca alone  or (cluster and client ca )

 if kafkauser creation uses client ca (which uses ceetmanager and not
Self Signed ) to sign,  can I still use the CN name for authorization ? 

When certificate and secret is renewed by certmanager , Will strimzi take necessary update action (like restart of pod, or anything which can affect due to cert renewal) even though we are using not the self signed cert?

**Comment by scholzj** [comment]
When you create a KafkaUser reosurce with authentication of `type: tls`, it uses the Clients CA to sign the user certificate. If you want to use CertManager to generate and sign the user certificates, you will need to provide the used CA as the Clients CA and then use wither the authentication `type: tls-external` which will mean that the User Operator will not issue any new certifciates (keep in mind that the subject of the certificates has to follow the User Operator conventions). Or you can disable the User Operator and manage the users your self.

**Reply by sreejesh123** [reply]
Thank you , it’s really helpful ,  to clarify  “used CA is client ca “ from your reply above  is change in user manefest ?  Where can I read more on how we can disable user operator and the subject standards please ? Appreciate if you can ping the url.

**Reply by scholzj** [reply]
> to clarify “used CA is client ca “ from your reply above is change in user manefest ?

Sorry, not sure I understand this question

> Where can I read more on how we can disable user operator

Just remove the whole `userOperator` block from the `Kafka` custom resource.

>  the subject standards please ?

I assume it will be somewhere in the docs, but I do not have any URL at hand. But it is real simple ... the subject should contain only the `CN` and it should be equal to the username / name of the `KafkaUser` custom resource. So for example, if you `KafkaUser` resource is called `jakub`, the certificate subject should contain `CN=jakub` and the username will be `CN=jakub`. If you choose to not use the User Operator, you get more freedom to set the certificate subject as you wish and all the subject segments will make it into the Kafka username.

**Reply by sreejesh123** [reply]
Thank you ,

**Reply by sreejesh123** [reply]
Sorry a very basic question, if I disable the useroperator can I still use Kafkauser kind to create user ?  Is there any example I can look at please if you know anyone  tried this in past?

**Reply by scholzj** [reply]
No, if you disable it, any KafkaUser resource will be ignored.

**Reply by scholzj** [reply]
You will need to use the Admin APIs to manage ACLs, quotas, SCRAM-SHA users etc.
